lib/writer/block-read-stream.js

```diff
@@ -91,7 +98,9 @@ class BlockReadStream extends stream.Readable {
 
         if (isTransient && (this.retries < this.maxRetries)) {
           this.retries += 1
-          this._read()
+          setTimeout(() => {
+            this._read()
```

Comment:
So in the BlockWriteStream the `_write` method sets a timeout to call `_write` again (if there was a transient error), but in the BlockReadStream it's actually the `_onRead` method that sets a timeout to call `_read` (if there was a transient error) ? I'm sure that's probably correct, and we'd have probably noticed already if it wasn't; this is just me being curious again ;-)

Reply:
Yup, because _onRead() is the callback that gets called once the _read() completes (or errors), while in the write stream the callback isn't pulled out due to scope deps, hence the difference.